feat: progressive web app

hoiekim opened the pull request:
Feature request: #328

This commit makes Docmost work as a [progressive web app](https://en.wikipedia.org/wiki/Progressive_web_app). Now users can add websites that host Docmost on their home screen and it provides native-application-like experience on iOS and Android.

Currently I deployed this changes to my self-hosted Docmost app. Try adding this website to home screen on your mobile devices: 

https://doc.hoie.kim

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/docmost/docmost?pullRequest=614) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/docmost/docmost?pullRequest=614) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/docmost/docmost?pullRequest=614) it.</sub>

robvanvolt commented:
Looks good - you could still adjust the "Backgorund" color to the background of the app, so you wouldn't have a sharp edge on the edge to the status bar. :)

aayushchugh commented:
@Philipinho Can you please review this?

2TAP2B commented:
Any updates here?

DenysMb commented:
> Feel free to send a PR. I will review. 

 _Originally posted by @Philipinho in [#328](https://github.com/docmost/docmost/issues/328#issuecomment-2377598884)_

There is a PR now, and it looks very simple. 
Could you review it, please?

Philipinho commented:
My issue with the PWA PRs so far is the many splash screen files.
If we can reduce that to the barest minimum, I am happy to revenue this PR again.

DenysMb commented:
@hoiekim What happens if no splash screen image are provided?

I don't have iPhone, but I would imagine that they would use a generic splash screen with the app logo if none were provided, is that right?

If so, is the splash screen really necessary?

hoiekim commented:
@DenysMb That's right - except there's no such thing as generic splash screen with the app logo. It'll be just blank black or white screen depending on the loading stage. I use dark mode so on my phone the app starts with black screen, then as the renderer(browser) starts it turns white until HTML elements get rendered.

Splash screen is necessary only for esthetic reasons. There's no right way to go about it but it's my understanding that having a properly color-themed splash screen gives users the impression that the app is mature, complete or well-designed. I know it sounds ambiguous and arbitrary so use your own judgment.

Although in this specific PR's situation I'd rather not insist my ego. I see that some people are uncomfortable with adding so much binary files and HTML head data. I'm proactively responding to that feedback and removed all splash screen and reverted the commits with those image files.

Thoroslives commented:
Any reason the merge hasnt occured?
